server/network/aoprotocol.py

```diff
@@ -1120,6 +1124,29 @@ def net_cmd_opBAN(self, args):
         """
         self.net_cmd_ct(['opban', '/ban {}'.format(args[0])])
 
+    @classmethod
+    def _change_to_escape_characters(cls, list_to_change: List[str]) -> List[str]:
+        escaped_names = []
+        for not_escaped_string in list_to_change:
+            if cls._contains_escape_character(not_escaped_string):
+                escaped_name = cls._add_escape_characters(not_escaped_string)
+            else:
+                escaped_name = not_escaped_string
+            escaped_names.append(escaped_name)
+        return escaped_names
+
+    @staticmethod
+    def _contains_escape_character(word: str) -> bool:
+        return any(char for char in ESCAPE_CHARACTERS.keys() if char in word)
+
+    @staticmethod
+    def _add_escape_characters(word_to_change: str) -> str:
+        for char in word_to_change:
+            escape_character = ESCAPE_CHARACTERS.get(char)
+            if escape_character:
+                word_to_change = word_to_change.replace(char, escape_character)
+        return word_to_change
```

Comment:
Why all this? Just run 4 regexes in a row and be done with it.

Reply:
I like the idea of abstracting things out into different decorator functions to try and make the code more readable. This lets someone read the code and have a better understanding rather than having to sit and ponder the 4 different regex. I strive to write code that reads almost like a sentence in a book. Something cohesive but abstracted and clean enough that anyone who looks at it should know what's going on without having to think about it too much. 

Abstracting these out also gives me the added benefit of possibly using these functions somewhere else in the class if need be. By breaking it out into these different parts, I can narrow down troubleshooting issues a little better and edit them in a way that is a minor code change rather than having to try and change an entire function. 

At the end of the day I think this is a coding preference. I guess you could run some profiler on it or something, but I think having them separated out like this reads a little better to me.

Reply:
I am telling you that using regex is a clean and readable solution taking less than six lines of code.